flocker/node/agents/test/test_blockdevice.py

```diff
@@ -2845,6 +2845,19 @@ def test_mountpoint_error(self):
         failure = self.failureResultOf(mount_result, OSError)
         self.assertEqual(ENOTDIR, failure.value.errno)
 
+    def test_mountpoint_permissions(self):
+        """
+        The mountpoint is world-writeable (since containers can run as any
+        user), and its parent is only accessible as current user (for
+        security).
+        """
+        mountroot = mountroot_for_test(self)
+        mountpoint = mountroot.child(b"mount-test")
+        self._run_success_test(mountpoint)
+        self.assertEqual((mountroot.getPermissions().shorthand(),
+                          mountpoint.getPermissions().shorthand()),
+                         ('rwx------', 'rwxrwxrwx'))
```

Comment:
:question: I can see how this will fix the problem, but those permissions (`rwxrwxrwx`) seem a bit too open when I see them represented that way.

 * User `root` owns the mount point directory.
 * A process in a container running as user `nobody` can create a file or a sub-directory and 
 * the default permissions on those will be determined by the umask of the process.
 * But *any user* on the node who knows the path to the mount point can delete all the files inside?

But that is prevented because the mount root directory permissions do not include user / group execute. Right?

Yeah, I see it now.

```
[~/tmp/foo]$ ls -la
total 0
drwx------. 1 richard richard   6 Jan 20 19:15 .
drwxrwxr-x. 1 richard richard 992 May 13 14:36 ..
drwxrwxr-x. 1 richard richard   6 Jan 20 19:15 bar
[~/tmp/foo]$ cd ../
[~/tmp]$ sudo chown root:root foo
[sudo] password for richard: 
[~/tmp]$ cd foo/
bash: cd: foo/: Permission denied
[~/tmp]$ cd foo/bar
bash: cd: foo/bar: Permission denied
[~/tmp]$ ls -l foo/bar/
ls: cannot access foo/bar/: Permission denied
[~/tmp]$ rm -rf foo/bar
rm: cannot remove ‘foo/bar’: Permission denied
[~/tmp]$ rm -rf foo/bar/baz
rm: cannot remove ‘foo/bar/baz’: Permission denied

```


So all those files and sub-directories are owned by user `nobody` on node1 who has UID 99

But what if `nobody` on node2 has a different UID?

I suppose that was always the case though.
Worth a followup issue?
Worth marking this bit of code for a future security audit?
Your call.